src/python/pants/rules/core/test.py

```diff
@@ -25,33 +24,37 @@ class Test(Goal):
 
 
 @console_rule
-def fast_test(console: Console, addresses: BuildFileAddresses) -> Test:
-  test_results = yield [Get(TestResult, Address, address.to_address()) for address in addresses]
+def fast_test(console: Console, targets: HydratedTargets, build_config: BuildConfiguration) -> Test:
```

Comment:
Although this horse is already out of the barn, it would be great not to rely on the `BuildConfiguration` like this... there is way too much "stuff" in there, and I'd be worried about getting into a `Context`-has-all-of-the-things situation. The `BuildConfiguration` is currently injected for the purposes of options parsing (I thought [here](https://github.com/pantsbuild/pants/blob/master/src/python/pants/engine/legacy/options_parsing.py)? but perhaps elsewhere), and I hope that we can shrink the requirements / surface area of those `@rule`s over time.

"Mirror" might not be the best name, but it would be good to expose some smaller purposeful type, even if that means installing:
```
@rule
def mirror(build_config: BuildConfiguration) -> Mirror:
   ..
```

Reply:
Fixed. PTAL.